eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -184,35 +195,83 @@ dRepAcceptedRatio RatifyEnv {reDRepDistr, reDRepState, reCurrentEpoch} gasDRepVo
 
     (yesStake, totalExcludingAbstainStake) = Map.foldlWithKey' accumStake (0, 0) reDRepDistr
 
+verifyPrev ::
+  forall era.
+  GovAction era ->
+  EnactState era ->
+  Bool
+verifyPrev act ens =
+  case act of
+    NoConfidence prev -> check ensCommitteeHPL prev
+    NewCommittee prev _ _ -> check ensCommitteeHPL prev
+    NewConstitution prev _ -> check ensConstitutionHPL prev
+    HardForkInitiation prev _ -> check ensProtVerHPL prev
+    ParameterChange prev _ -> check ensCurPParamsHPL prev
+    TreasuryWithdrawals {} -> True
+    InfoAction {} -> True
+  where
+    check ::
+      Lens' (EnactState era) (HashProtected r a (EraCrypto era)) ->
+      StrictMaybe (PrevGovActionId r (EraCrypto era)) ->
+      Bool
+    check h prev = hpId (ens ^. h) == prev
```

Comment:
You are really not gaining any clarity or reduction in amount of code by adding a helper `check` function like this :slightly_smiling_face: : 
```suggestion
    NoConfidence prev -> hpId (ens ^. ensCommitteeHPL) == prev
    NewCommittee prev _ _ -> hpId (ens ^. ensCommitteeHPL) == prev
    NewConstitution prev _ -> hpId (ens ^. ensConstitutionHPL) == prev
    HardForkInitiation prev _ -> hpId (ens ^. ensProtVerHPL) == prev
    ParameterChange prev _ -> hpId (ens ^. ensCurPParamsHPL) == prev
    TreasuryWithdrawals {} -> True
    InfoAction {} -> True
```